chore(Authoring): Convert WISE link authoring to Angular

geoffreykwan opened the pull request:
## Changes

Converted WISELinkAuthoringController to Angular.

## Test

Make sure the WISE link authoring works in the HTML component.

Closes #1360

geoffreykwan committed:
chore(Authoring): Convert WISE link authoring to Angular

#1360

hirokiterashima commented on src/assets/wise5/authoringTool/wise-link-authoring-dialog/wise-link-authoring-dialog.component.ts:
```diff
@@ -0,0 +1,75 @@
+  styleUrls: ['./wise-link-authoring-dialog.component.scss']
+})
+export class WiseLinkAuthoringDialogComponent {
+  items: any[];
+  wiseLinkComponentId: string = '';
+  wiseLinkNodeId: string = '';
+  wiseLinkText: string = '';
+  wiseLinkType: string = 'link';
```
Add private/protected modifiers?

hirokiterashima commented on src/assets/wise5/authoringTool/wise-link-authoring-dialog/wise-link-authoring-dialog.component.ts:
```diff
@@ -0,0 +1,75 @@
+    return this.projectService.getComponents(nodeId);
+  }
+
+  protected createWISELink(): void {
```
The name create() is sufficient?

hirokiterashima commented on src/assets/wise5/authoringTool/wise-link-authoring-dialog/wise-link-authoring-dialog.component.ts:
```diff
@@ -0,0 +1,75 @@
+    }
+  }
+
+  protected cancelWISELinkAuthoring(): void {
```
The name cancel() is sufficient?

hirokiterashima reviewed (approved):
LGTM.

Before merge, can you remove UpgradeModule dependency in WiseAuthoringTinymceEditorComponent? It's no longer used.

I also added some code improvement suggestions inline for your consideration.

geoffreykwan committed:
chore(Authoring): Clean up code

Add variable modifiers, remove unused dependency, and change function
names.

#1360

hirokiterashima commented:
:tada: This PR is included in version 5.107.0 :tada:

The release is available on [GitHub release](https://github.com/WISE-Community/WISE-Client/releases/tag/v5.107.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
